sdk/python/pulumi_kubernetes/helm/v2/helm.py

```diff
@@ -294,52 +295,72 @@ def __init__(self,
         self.path = path
 
 
+def _run_helm_cmd(cmd: List[Union[str, bytes]]):
+    output = subprocess.run(
+        cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True, check=True)
+    yaml_str: str = output.stdout
+    return yaml_str
+
+
+def _cleanup_temp_dir(all_config: Tuple[TextIO, Union[bytes, str], pulumi.Output]):
+    file, chart_dir, _ = all_config
+
+    file.close()
+    shutil.rmtree(chart_dir)
+
+
 def _parse_chart(all_config: Tuple[str, Union[ChartOpts, LocalChartOpts], pulumi.ResourceOptions]) -> pulumi.Output:
     release_name, config, opts = all_config
 
     # Create temporary directory and file to hold chart data and override values.
-    with NamedTemporaryFile() as overrides:
-        with TemporaryDirectory() as chart_dir:
-            if isinstance(config, ChartOpts):
-                chart_to_fetch = f'{config.repo}/{config.chart}' if config.repo else config.chart
-
-                # Configure fetch options.
-                fetch_opts_dict = {}
-                if config.fetch_opts is not None:
-                    fetch_opts_dict = {k: v for k, v in vars(config.fetch_opts).items() if v is not None}
-                fetch_opts_dict["destination"] = chart_dir
-                if config.version is not None:
-                    fetch_opts_dict["version"] = config.version
-                fetch_opts = FetchOpts(**fetch_opts_dict)
-
-                # Fetch the chart.
-                _fetch(chart_to_fetch, fetch_opts)
-                fetched_chart_name = os.listdir(chart_dir)[0]
-                chart = os.path.join(chart_dir, fetched_chart_name)
-            else:
-                chart = config.path
-
-            default_values = os.path.join(chart, 'values.yaml')
-
-            # Write overrides file.
-            vals = config.values if config.values is not None else {}
-            data = json.dumps(vals).encode('utf-8')
-            overrides.write(data)
-            overrides.flush()
-
-            namespace_arg = ['--namespace', config.namespace] if config.namespace else []
-
-            # Use 'helm template' to create a combined YAML manifest.
-            cmd = ['helm', 'template', chart, '--name', release_name,
-                   '--values', default_values, '--values', overrides.name]
-            cmd.extend(namespace_arg)
-
-            output = subprocess.run(
-                cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True, check=True)
-            yaml_str: str = output.stdout
-
-            # Parse the manifest and create the specified resources.
-            return _parse_yaml_document(yaml.safe_load_all(yaml_str), opts, config.transformations)
+    # Note: Using context managers for this was not playing nicely with the async Outputs, so we're using
+    # the lower-level APIs and manually cleaning up once we're done.
```

Comment:
We should say what was actually happening. Also -- are we actually cleaning these up when we're done? I don't see a delete anywhere.

Reply:
Will update the comment. Delete is here: https://github.com/pulumi/pulumi-kubernetes/pull/670/files#diff-056a55e1e4e98cb2309204a4b0dbd46fR362